### PR TITLE
CHE-3209 add import and configure batch of projects on stack testing

### PR DESCRIPTION
**Pull request by olexii4**
What does this PR do?

Add import and configure batch of projects on stack testing

What issues does this PR fix or reference?

#3209 

selection_051
selection_053

Signed-off-by: Oleksii Orel <oorel@codenvy.com>

### CODE REVIEW

**Review by akurinnoy — APPROVED** [review]

File: dashboard/src/app/stacks/stack-details/select-template/select-template.controller.ts [hunk 1]
```diff
@@ -71,9 +71,10 @@ export class SelectTemplateController {
    */
   startTest(): void {
     let stack: che.IStack = angular.copy(this.stack);
+    /* tslint:disable */
```
Comment (benoitf) [inline dashboard/src/app/stacks/stack-details/select-template/select-template.controller.ts:74]: question: why do we need to disable tslint ?

File: dashboard/src/app/stacks/stack-details/stack.controller.ts [hunk 2]
```diff
@@ -101,7 +103,16 @@ export class StackController {
 
     $window.addEventListener('message', (event: {data: string}) => {
       if ('show-ide' === event.data) {
+        if (this.tmpWorkspaceId) {
+          //TODO remove this delay after fixing time of update tree
```
Comment (benoitf) [inline dashboard/src/app/stacks/stack-details/stack.controller.ts]: is there an open github issue ?

File: dashboard/src/app/stacks/stack-details/stack.controller.ts [hunk 3]
```diff
@@ -347,13 +358,86 @@ export class StackController {
   }
 
   /**
+   * Add projects sequentially by iterating on the number of the projects.
+   * @param workspaceId{string} - the ID of the workspace to use for adding commands
+   * @param projects{Array<any>} - the array to follow
+   * @param index{number} - the index of the array of commands
+   * @param deferred{ng.IDeferred<any>}
+   */
+  updateProjects(workspaceId: string, projects: Array<any>, index: number, deferred: ng.IDeferred<any>) {
```
Comment (benoitf) [inline dashboard/src/app/stacks/stack-details/stack.controller.ts]: return type is void ?

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 2 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: Not yet. It is a time for project type update(from API). I should investigate it more.

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 3 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: Yes. I will fix

File: dashboard/src/app/stacks/stack-details/select-template/select-template.controller.ts (hunk 1 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/select-template/select-template.controller.ts:74]: forbidden bitwise operation
https://github.com/eclipse/che/blob/master/dashboard/tslint.json#L46

File: dashboard/src/app/workspaces/workspace-details/workspace-details.controller.ts [hunk 4]
```diff
@@ -398,11 +398,13 @@ export class WorkspaceDetailsController {
...
    */
   generateWorkspaceName(): string {
-    let name,
-        iterations = 100;
+    let name: string,
+        iterations: number = 100;
     while (iterations--) {
-      name = 'wksp-' + (('0000' + (Math.random() * Math.pow(36, 4) << 0).toString(36)).slice(-4)); // jshint ignore:line
-      if (!this.usedNamesList.indexOf(name) >= 0) {
+      /* tslint:disable */
```
[the remark this answers is not in the document]
Comment (olexii4) [reply dashboard/src/app/workspaces/workspace-details/workspace-details.controller.ts:418]: forbidden bitwise operation
https://github.com/eclipse/che/blob/master/dashboard/tslint.json#L46

**Review by benoitf — APPROVED** [review]

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 2 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: fixed

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 3 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: fixed

**Pull request merged**